app/controllers/RemoveAreYouSureController.scala

```diff
@@ -80,11 +80,15 @@ class RemoveAreYouSureController @Inject() (
                 .bindFromRequest()
                 .fold(
                   formWithErrors => Future.successful(BadRequest(view(formWithErrors, institutionToRemove.FIID, institutionToRemove.FIName))),
-                  value =>
+                  value => {
+                    if (value) {
+                      financialInstitutionsService.removeFinancialInstitution(institutionToRemove)
+                    }
```

Comment:
This can give user a wrong feedback, in case the remove fi is failed or for very least, user gets redirected to your-fis page, the fi may not be removed yet. In both scenarios it isn't a good response. We probably should wait for the remove fi to be completed before redirecting and also in case of error, it will automatically redirect user to error page:

```
                  value => {
                    for {
                      _ <- if (value) financialInstitutionsService.removeFinancialInstitution(institutionToRemove) else Future.successful(())
                      updatedAnswers <- Future.fromTry(request.userAnswers.set(RemoveAreYouSurePage, value))
                      _              <- sessionRepository.set(updatedAnswers)
                    } yield Redirect(navigator.nextPage(RemoveAreYouSurePage, NormalMode, updatedAnswers))
                  }
```